src/psyclone/tests/psyir/tools/dependency_tools_test.py

```diff
@@ -172,25 +175,115 @@ def test_arrays_parallelise(parser):
+                                 end do
+                                 end program test''')
+    prog = parser(reader)
+    psy = PSyFactory("nemo", distributed_memory=False).create(prog)
+    loops = psy.invokes.get("test").schedule
+    dep_tools = DependencyTools(["levels", "lat"])
+
+    var_info0 = VariablesAccessInfo(loops[0])
+    var_info1 = VariablesAccessInfo(loops[1])
+    jj_symbol = loops.scope.symbol_table.lookup("jj")
+
+    # Test 1: provide access pattern of different variables:
+    sig_a = Signature("a")
+    sig_b = Signature("b")
+    sig_c = Signature("c")
+    with pytest.raises(InternalError) as err:
+        _ = dep_tools.array_accesses_consistent(jj_symbol,
+                                                [var_info0[sig_a],
+                                                 var_info1[sig_b],
+                                                 var_info1[sig_c]])
+    assert "Inconsistent signature provided in 'array_accesses_consistent'. " \
+           "Expected all access to be for 'a', but also got 'b,c'."\
+           in str(err.value)
+
+    # Test 2: provide a consistent list of accesses.
+    # Check number of messages and returned access array for correctness.
+    a_access_1st_loop = var_info0[sig_a]
+    a_access_2nd_loop = var_info1[sig_a]
+    all_ind = []
+    consistent = dep_tools.array_accesses_consistent(jj_symbol,
+                                                     [a_access_1st_loop,
+                                                      a_access_2nd_loop],
+                                                     all_ind)
+    assert consistent
+    assert len(dep_tools.get_all_messages()) == 0
+    assert len(all_ind) == 3
+    assert all_ind[0] == a_access_1st_loop[0].component_indices[(0, 1)]
+    assert all_ind[1] == a_access_2nd_loop[0].component_indices[(0, 1)]
+    assert all_ind[2] == a_access_2nd_loop[1].component_indices[(0, 1)]
+
+    # Test 3: provide a single instance (not a list).
+    all_ind = []
+    consistent = dep_tools.array_accesses_consistent(jj_symbol,
+                                                     a_access_1st_loop,
+                                                     all_ind)
+    assert consistent
+    assert len(dep_tools.get_all_messages()) == 0
+    assert all_ind == [a_access_1st_loop[0].component_indices[(0, 1)]]
+
+    # Test 4: trigger an error.
+    var_info2 = VariablesAccessInfo(loops[2])
+    a_access_3rd_loop = var_info2[sig_a]
+    all_ind = []
+    consistent = dep_tools.array_accesses_consistent(jj_symbol,
+                                                     [a_access_1st_loop,
+                                                      a_access_3rd_loop],
+                                                     all_ind)
+    assert not consistent
+    assert len(dep_tools.get_all_messages()) == 1
+    assert "Variable 'a' is written to and the loop variable 'jj' is used " \
+           "differently: a(ji,jj) and a(jj,ji)." \
```

Comment:
s/is used differently/is used in different index locations/ perhaps?

Reply:
(I can see the value of having actual Fortran in the error message here though.)

Reply:
I'm glad that you agree and this wasn't all wasted :)